pkg/logql/log/metrics_extraction.go

```diff
@@ -132,12 +137,18 @@ func LabelExtractorWithStages(
 		groups = append(groups, labelName)
 		sort.Strings(groups)
 	}
+	var expectedLabels []string
+	if !without {
+		expectedLabels = append(postFilter.RequiredLabelNames(), groups...)
+		expectedLabels = append(expectedLabels, postFilter.RequiredLabelNames()...)
```

Comment:
Why are we adding `postFilter.RequiredLabelNames()` twice?

Reply:
Not sure what this is, bad copy 🍝  may be.

I'll fix this up.